api/web3server_utils.go

```diff
@@ -117,7 +118,19 @@ func (svr *web3Handler) getTransactionFromActionInfo(blkHash hash.Hash256, selp
 	if err != nil {
 		return nil, err
 	}
-	tx, err := action.RawTxToSignedTx(ethTx, svr.coreService.EVMNetworkID(), selp.Signature())
+	var (
+		signer   types.Signer
+		encoding = iotextypes.Encoding(selp.Encoding())
+	)
+	if encoding != iotextypes.Encoding_IOTEX_PROTOBUF {
+		signer, err = action.NewEthSigner(encoding, svr.coreService.EVMNetworkID())
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		signer = types.HomesteadSigner{}
+	}
```

Comment:
revert this change in this pr and keep only refactoring part?

Reply:
make sense, this should be part of the next (#3965)